PWGHF/D2H/TableProducer/dataCreatorCharmHadPiReduced.cxx

```diff
@@ -109,47 +110,50 @@ enum WrongCollisionType : uint8_t {
 struct HfDataCreatorCharmHadPiReduced {
   // Produces AOD tables to store track information
   // collision related tables
-  Produces<aod::HfRedCollisions> hfReducedCollision;
-  Produces<aod::HfRedCollCents> hfReducedCollCentrality;
-  Produces<aod::HfRedQvectors> hfReducedQvector;
-  Produces<aod::HfRedCollExtras> hfReducedCollExtra;
-  Produces<aod::HfOrigColCounts> hfCollisionCounter;
-  // Pi bachelor related tables
-  Produces<aod::HfRedTrackBases> hfTrackPion;
-  Produces<aod::HfRedTracksCov> hfTrackCovPion;
-  Produces<aod::HfRedTracksPid> hfTrackPidPion;
-  // charm hadron related tables
-  Produces<aod::HfRed2Prongs> hfCand2Prong;
-  Produces<aod::HfRed2ProngsCov> hfCand2ProngCov;
-  Produces<aod::HfRed2ProngsMl> hfCand2ProngMl;
-  Produces<aod::HfRed3Prongs> hfCand3Prong;
-  Produces<aod::HfRed3ProngsCov> hfCand3ProngCov;
-  Produces<aod::HfRed3ProngsMl> hfCand3ProngMl;
-  // PID tables for charm-hadron candidate daughter tracks
-  Produces<aod::HfRedPidDau0s> hfCandPidProng0;
-  Produces<aod::HfRedPidDau1s> hfCandPidProng1;
-  Produces<aod::HfRedPidDau2s> hfCandPidProng2;
-
-  // B-hadron config and MC related tables
-  Produces<aod::HfCandB0Configs> rowCandidateConfigB0;
-  Produces<aod::HfMcRecRedDpPis> rowHfDPiMcRecReduced;
-  Produces<aod::HfMcCheckDpPis> rowHfDPiMcCheckReduced;
-  Produces<aod::HfMcGenRedB0s> rowHfB0McGenReduced;
-
-  Produces<aod::HfCandBpConfigs> rowCandidateConfigBplus;
-  Produces<aod::HfMcRecRedD0Pis> rowHfD0PiMcRecReduced;
-  Produces<aod::HfMcCheckD0Pis> rowHfD0PiMcCheckReduced;
-  Produces<aod::HfMcGenRedBps> rowHfBpMcGenReduced;
-
-  Produces<aod::HfCandBsConfigs> rowCandidateConfigBs;
-  Produces<aod::HfMcRecRedDsPis> rowHfDsPiMcRecReduced;
-  Produces<aod::HfMcCheckDsPis> rowHfDsPiMcCheckReduced;
-  Produces<aod::HfMcGenRedBss> rowHfBsMcGenReduced;
-
-  Produces<aod::HfCandLbConfigs> rowCandidateConfigLb;
-  Produces<aod::HfMcRecRedLcPis> rowHfLcPiMcRecReduced;
-  Produces<aod::HfMcCheckLcPis> rowHfLcPiMcCheckReduced;
-  Produces<aod::HfMcGenRedLbs> rowHfLbMcGenReduced;
+  struct : ProducesGroup {
+    Produces<aod::HfRedCollisions> hfReducedCollision;
+    Produces<aod::HfRedCollCents> hfReducedCollCentrality;
+    Produces<aod::HfRedQvectors> hfReducedQvector;
+    Produces<aod::HfRedCollExtras> hfReducedCollExtra;
+    Produces<aod::HfOrigColCounts> hfCollisionCounter;
+    // Pi bachelor related tables
+    Produces<aod::HfRedTrackBases> hfTrackPion;
+    Produces<aod::HfRedTracksCov> hfTrackCovPion;
+    Produces<aod::HfRedTracksPid> hfTrackPidPion;
+    // charm hadron related tables
+    Produces<aod::HfRed2Prongs> hfCand2Prong;
+    Produces<aod::HfRed2ProngsCov> hfCand2ProngCov;
+    Produces<aod::HfRed2ProngsMl> hfCand2ProngMl;
+    Produces<aod::HfRed3Prongs> hfCand3Prong;
+    Produces<aod::HfRed3ProngsCov> hfCand3ProngCov;
+    Produces<aod::HfRed3ProngsMl> hfCand3ProngMl;
+    Produces<aod::HfRedB0SoftPi> hfCandDstarSoftPi;
```

Comment:
For the soft pion, we need the corresponding track tables as for the bachelor:
```
    Produces<aod::HfRedTrackBases> hfTrackPion;
    Produces<aod::HfRedTracksCov> hfTrackCovPion;
```
Without the track parameters otherwise we cannot re-reconstruct the B vertex from the derived data